src/pip/_internal/exceptions.py

```diff
@@ -323,12 +323,14 @@ class UnsupportedWheel(InstallationError):
 class InvalidWheel(InstallationError):
     """Invalid (e.g. corrupt) wheel."""
 
-    def __init__(self, location: str, name: str):
+    def __init__(self, location: str, name: str, context: Optional[str] = None):
         self.location = location
         self.name = name
+        self.context = context
 
     def __str__(self) -> str:
-        return f"Wheel '{self.name}' located at {self.location} is invalid."
+        suffix = f" ({self.context})" if self.context else "."
```

Comment:
```suggestion
        suffix = f" ({self.context})." if self.context else "."
```

(or better, keep the trailing full stop in the return statement rather than here, but I don't think github will let me add 2-line suggestions...)

Reply:
Done!